### PR TITLE
unit test for write method

### PR DESCRIPTION
**Pull request by adamplansky**

### CODE REVIEW

File: curly/main.go [hunk 1]
```diff
@@ -121,51 +121,56 @@ func logErr(err error) {
...
+	prefix := p
+	return func() (io.WriteCloser, error) {
+		i++
+		return os.Create(fmt.Sprintf("%s.%d", prefix, i))
+	}
+}
+
+func NewChunked(prefix string) (*Chunked, error) {
+	floppyFunc := fileFloppyFn(prefix)
+	w, err := floppyFunc()
```
Comment (sev3ryn) [inline curly/main.go:141]: nit: code will be nice if you also do first invocation of `floppyFunc` lazily in Write method

File: curly/main.go [hunk 2]
```diff
@@ -121,51 +121,56 @@ func logErr(err error) {
...
-	_, err := c.F.Write(p[:off])
+	off := c.maxSize - c.size
+	n, err := c.w.Write(p[:off])
 	if err != nil {
-		return 0, err
+		return n, err
 	}
-	c.F, err = os.Create(fmt.Sprintf("%s.%d", c.Prefix, c.idx))
-	c.idx++
+	c.w, err = c.floppyCreateFn()
```
Comment (sev3ryn) [inline curly/main.go:165]: I don't see you closing files in ChunkedWriter

File: curly/main.go [hunk 3]
```diff
@@ -121,51 +121,56 @@ func logErr(err error) {
...
 	if err != nil {
-		return off, err
+		return n, err
 	}
-	c.fileSize = pLen - off
-	_, err = c.F.Write(p[off:])
+	c.size = len(p) - off
+	n, err = c.w.Write(p[off:])
```
Comment (sev3ryn) [inline curly/main.go:170]: What if `len(p) > 2*maxSize`?

Also add such test case

File: curly/main_test.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+				maxSize:        tt.maxSize,
+				floppyCreateFn: floppyFn,
+			}
+			for _, p := range tt.p {
+				_, err = c.Write(p)
+			}
+
+			assert.NoError(t, err)
+			for i := range result {
+				assert.Equal(t, tt.want[i], result[i].Buf)
```
Comment (sev3ryn) [inline curly/main_test.go:73]: nit: take a look on https://pkg.go.dev/github.com/google/go-cmp/cmp#Diff, output is nicer also it compares unexported fields

File: curly/main_test.go [hunk 5]
```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"github.com/stretchr/testify/assert"
+	"io"
+	"testing"
+)
+
+func ioFloppyCreateFn() ([]*testChunked, func() (io.WriteCloser, error)) {
+	ww := make([]*testChunked, 0)
```
Comment (sev3ryn) [inline curly/main_test.go:10]: nit: bytes.Buffer instead of testChunked will save you few lines of code :)

**Review by sev3ryn — COMMENTED** [review]
There are two important issues which make me not approve this PR for now. Also there are few nits that do not require action but are worth to consider

**Comment by adamplansky** [comment]
All issues should be solved in latest version.

File: curly/chunker_test.go [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+var _ Chunker = &testChunked{}
+
+type testChunked struct {
+	multiBuffer []*bytes.Buffer
+	buf         *bytes.Buffer
```
Comment (sev3ryn) [inline curly/chunker_test.go:14]: nit: imho `buf` is not needed you can always access it as multiBuffer[len(multiBuffer)-1]

**Comment by sev3ryn** [comment]
LGTM, good to merge

**Pull request merged**